bin/benchmark/Sources/benchmark/Subcommands/Measure.swift

```diff
@@ -0,0 +1,333 @@
+                    let indexDirectory = doccOutputLocation.appendingPathComponent(NodeURLGenerator.Path.indexFolderName, isDirectory: true)
+                    if FileManager.default.fileExists(atPath: indexDirectory.path),
+                       let newResult = Benchmark.IndexDirectoryOutputSize(indexDirectory: indexDirectory).result
+                    {
+                        recomputedMissingMetrics.append(
+                            .init(
+                                id: Benchmark.IndexDirectoryOutputSize.identifier,
+                                displayName: Benchmark.IndexDirectoryOutputSize.displayName,
+                                value: newResult
+                            )
+                        )
+                    }
+                }
+                
+                benchmarkResult = BenchmarkResults(
+                    platformName: benchmarkResult.platformName,
+                    timestamp: benchmarkResult.timestamp,
+                    doccArguments: benchmarkResult.doccArguments,
+                    unorderedMetrics: benchmarkResult.metrics + recomputedMissingMetrics
+                )
+            }
+            
+            try benchmarkSeries.add(benchmarkResult)
+        }
+        
+        return benchmarkSeries
+    }
+    
+    static func writeResults(_ benchmarkSeries: BenchmarkResultSeries, to outputLocation: URL) throws {
+        let parentDirectory = outputLocation.deletingLastPathComponent()
+        if !FileManager.default.fileExists(atPath: parentDirectory.path) {
+            try FileManager.default.createDirectory(at: parentDirectory, withIntermediateDirectories: true)
+        }
+        
+        let encoder = JSONEncoder()
+        encoder.outputFormatting = [.prettyPrinted, .sortedKeys]
+        let outputData = try encoder.encode(benchmarkSeries)
+        try outputData.write(to: outputLocation)
+    }
+}
+
+private let envURL = URL(fileURLWithPath: "/usr/bin/env")
+
+private func runTask(_ url: URL, directory: URL? = nil, arguments: [String] = [], environment: [String: String] = [:]) throws {
+    let task = Process()
+    if #available(macOS 10.13, *) {
+        task.currentDirectoryURL = directory
+    } else if let directoryPath = directory?.path {
+        task.currentDirectoryPath = directoryPath
+    }
+    if #available(macOS 10.13, *) {
+        task.executableURL = url
+    } else {
+        task.launchPath = url.path
+    }
+    task.arguments = arguments
+    task.environment = ProcessInfo.processInfo.environment.merging(environment, uniquingKeysWith: +)
+    
+    task.launch()
+    task.waitUntilExit()
+    
+    
+    guard task.terminationStatus == 0 else {
+        throw Measure.Error.subtaskTerminated(task.terminationStatus, task.terminationReason)
+    }
+}
+
+private func waitUntilFairThermalState(minimumWait: Int) throws {
+    print("Preparing for next stage", terminator: " ")
```

Comment:
Nit
```suggestion
    print("Preparing for next stage", terminator: " ")
```

Reply:
I can move this space to the end of the string instead of the terminator.

Reply:
Addressed in c34eaaa078ea63733969ca2e21f5449bf7d336ef